fork_daemon.php

```diff
@@ -1729,7 +1729,8 @@ protected function process_work_unit($bucket = self::DEFAULT_BUCKET)
 			if ($this->child_single_work_item[$bucket])
 			{
 				// break out identifier and unit
-				list($child_identifier, $child_work_unit) = each($child_work_units);
+				list($child_identifier, $child_work_unit) = current($child_work_units);
```

Comment:
`each` is pulling both the key and value, using `current` is just going to pull the value which would make both items in the list empty.

You can split this into multiple calls though

```php
$child_identifier = key($child_work_units);
$child_work_unit = current($child_work_units);
next($child_work_units);
```